pennylane/qnode_new/qnode.py

```diff
@@ -0,0 +1,935 @@
+    They are useful for providing data or 'placeholders' to the quantum function.
+    In *mutable* nodes the auxiliary parameters can undergo any kind of classical
+    processing, and also appear in the ``wires`` argument of the Operations.
+    In *immutable* nodes they are subject to the same restrictions as the positional parameters.
+
+    Parameters that have default values are interpreted as auxiliary parameters. They *must* be
+    given using the keyword syntax.
+    TODO ALTERNATIVELY: Keyword-only parameters are interpreted as auxiliary parameters.
+
+The quantum function cannot be executed on its own. Instead, a :class:`~.QNode` object must be
+created, which wraps the function and binds it to a device. The QNode can then be used to evaluate
+the quantum circuit function on the particular device.
+
+For example:
+
+.. code-block:: python
+
+    device = qml.device('default.qubit', wires=2)
+    qnode1 = qml.QNode(my_quantum_function, device)
+    result = qnode1(np.pi/4, 0.7)
+
+.. note::
+
+        The :func:`~pennylane.decorator.qnode` decorator is provided as a convenience
+        to automate the process of creating quantum nodes. The decorator is used as follows:
+
+        .. code-block:: python
+
+            @qml.qnode(device)
+            def my_quantum_function(x, y):
+                qml.RZ(x, wires=0)
+                qml.CNOT(wires=[0,1])
+                qml.RY(y, wires=1)
+                return qml.expval(qml.PauliZ(0))
+
+            result = my_quantum_function(np.pi/4, 0.7)
+
+
+.. currentmodule:: pennylane.qnode
+
+
+Classes
+-------
+
+.. autosummary::
+   QNode
+   JacobianQNode
+   ParSig
+   ParDep
+   AutogradMixin
+
+
+QNode methods
+-------------
+
+.. currentmodule:: pennylane.qnode_new.qnode.QNode
+
+.. autosummary::
+   __call__
+   evaluate
+   evaluate_obs
+   _append_op
+   _construct
+   _check_circuit
+   _sort_args
+   _check_args
+   _default_args
+   _set_variables
+   _op_descendants
+
+
+JacobianQNode methods
+---------------------
+
+.. currentmodule:: pennylane.qnode_new.jacobian.JacobianQNode
+
+.. autosummary::
+   jacobian
+   _best_method
+   _pd_finite_diff
+   _pd_parameter_shift
+   _pd_analytic_var
+   _transform_observable
+
+.. currentmodule:: pennylane.qnode_new.qnode
+
+Exceptions
+----------
+
+.. autosummary::
+   QuantumFunctionError
+
+Code details
+~~~~~~~~~~~~
+"""
+
+from functools import wraps, lru_cache
+from collections.abc import Sequence
+from collections import namedtuple, OrderedDict
+import inspect
+import itertools
+
+import numpy as np
+from scipy import linalg
+
+import pennylane.measure as pm
+import pennylane.operation as plo
+import pennylane.ops as plops
+from pennylane.utils import _flatten, unflatten, expand
+from pennylane.circuit_graph import CircuitGraph, _is_observable
+from pennylane.variable import Variable
+from pennylane.qnode import QNode as QNode_old, QuantumFunctionError
+
+
+
+def qnode(device, *, mutable=True, properties=None):
```

Comment:
The decorator is here only temporarily. The old one does not work with the new QNode, and some tests required the decorator. In the final PR, when the old QNode is replaced, this should replace the old decorator in `decorator.py`.